manager/qemu/persistence.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Ultraviolet
+// SPDX-License-Identifier: Apache-2.0
+package qemu
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+type VMState struct {
+	ID     string
+	Config Config
+	PID    int
+}
+
+type FilePersistence struct {
+	dir  string
+	lock sync.Mutex
+}
+
+// Persistence is an interface for saving and loading VM states.
+//
+//go:generate mockery --name Persistence --output=./mocks --filename persistence.go --quiet --note "Copyright (c) Ultraviolet \n // SPDX-License-Identifier: Apache-2.0"
+type Persistence interface {
+	SaveVM(state VMState) error
+	LoadVMs() ([]VMState, error)
+	DeleteVM(id string) error
+}
+
+func NewFilePersistence(dir string) (Persistence, error) {
+	if err := os.MkdirAll(dir, 0o755); err != nil {
+		return nil, err
+	}
+	return &FilePersistence{dir: dir}, nil
+}
+
+func (fp *FilePersistence) SaveVM(state VMState) error {
```

Comment:
What is this doing? Saving VM on disk?

Never leave RAM, only RAM is in TEE!

Reply:
this is on the host(manager) not in TEE